lib/srv/authhandlers.go

```diff
@@ -742,7 +1027,62 @@ func (h *AuthHandlers) authorityForCert(caType types.CertAuthType, key ssh.Publi
 	return ca, nil
 }
 
-// isProxy returns true if it's a regular SSH proxy.
-func (h *AuthHandlers) isProxy() bool {
-	return h.c.Component == teleport.ComponentProxy
+func durationToGoDuration(d *durationpb.Duration) time.Duration {
+	// nil or "zero" Timestamps are mapped to Go's zero time (0-0-0 0:0.0) instead
+	// of unix epoch. The latter avoids problems with tooling (eg, Terraform) that
+	// sets structs to their defaults instead of using nil.
+	if d == nil || (d.Seconds == 0 && d.Nanos == 0) {
```

Comment:
```suggestion
	if d.GetSeconds() == 0 && d.GetNanos() == 0 {
```

Reply:
Wait what's the point of this? When do we need the distinction between a nil durationpb and a zero durationpb?

Reply:
The `timestamppb.Timestamp` and `time.Time` have incompatible zero values, resulting in the following behavior:

```text
(*timestamppb.Timestamp)(nil).AsTime().IsZero() -> false
timestamppb.New(time.Time{}) == nil -> false
```

Some of the initial `lib/decision` stuff we had in earlier PRs needed zero-preserving conversion, and I've just kinda been sticking with it for decision related stuff.  It's probably not necessary everywhere, but it seemed like a safer bet.

Reply:
That makes sense for timestamps, but this is a duration.